azure_identity_access_token_provider.go

```diff
@@ -127,15 +133,22 @@ func (p *AzureIdentityAccessTokenProvider) GetAllowedHostsValidator() *absauth.A
 }
 
 func isLocalhost(host string) bool {
-	if strings.HasPrefix(host, "localhost") {
-		remainder := strings.TrimPrefix(host, "localhost")
-		return remainder == "" || strings.HasPrefix(remainder, ":")
+	normalizedHost := strings.ToLower(host)
+	if strings.HasPrefix(normalizedHost, Localhost) {
```

Comment:
::1 is also valid. You could make the const an array and iterate over the array instead.